types/nightwatch/index.d.ts

```diff
@@ -2031,6 +2602,38 @@ export interface ClientCommands {
             | ((client: NightwatchAPI, done: () => void) => void),
     ): this;
 
+    /**
+     * Waits for a condition to evaluate to a "truthy" value. The condition may be specified by any function which
+     * returns the value to be evaluated or a Promise to wait for.
+     *
+     * An optional wait time can be specified, otherwise the global waitForConditionTimeout value will be used.
+     *
+     * @example
+     * describe('waitUntil Example', function() {
+     *   it('demo Test', function(browser) {
+     *     browser
+     *       .url('https://nightwatchjs.org)
+     *       .waitUntil(async function() {
+     *         const title = await this.execute(function() {
```

Comment:
the `this` is not specified in the callback function in the `conditionFn` definition below.
Something like this would be required:
```ts
    waitUntil(
        conditionFn:
            | ((this: NightwatchAPI) => undefined | Promise<any>)
            | ((this: NightwatchAPI, done: () => void) => void)
            | ((this: NightwatchAPI, client: NightwatchAPI, done: () => void) => void),
        waitTimeMs?: number,
        retryInterval?: number,
        callback?: (this: NightwatchAPI) => void,
    ): this;
```